src/algorithm/convexhull.rs

```diff
@@ -0,0 +1,199 @@
+    hull
+}
+
+// recursively calculate the convex hull of a subset of points
+fn hull_set<T>(p_a: &Point<T>, p_b: &Point<T>, set: &mut Vec<Point<T>>, hull: &mut Vec<Point<T>>)
+    where T: Float
+{
+    let insert_position = hull.iter().position(|r| r == p_b).unwrap();
+    if set.is_empty() {
+        return;
+    }
+    if set.len() == 1 {
+        hull.insert(insert_position, set[0]);
+        set.remove(0);
+        return;
+    }
+    let mut furthest_distance = Float::min_value();
+    let mut furthest_idx = <usize>::min_value();
+    for (idx, point) in set.iter().enumerate() {
+        let current_distance = pseudo_distance(p_a, p_b, point);
+        if current_distance > furthest_distance {
+            furthest_distance = current_distance;
+            furthest_idx = idx
+        }
+    }
+    // move Point at furthest_point from set into hull
+    let furthest_point = set[furthest_idx];
+    hull.insert(insert_position, set[furthest_idx]);
+    set.remove(furthest_idx);
+    // Determine points to the left of A, furthest_point
+    let mut left_ap: Vec<Point<T>> = vec![];
+    let mut left_pb: Vec<Point<T>> = vec![];
+    for point in set.iter() {
+        if point_location(p_a, &furthest_point, point) {
+            left_ap.push(*point);
+        }
+    }
+    // Determine points to the left of furthest_point, B
+    for point in set.iter() {
+        if point_location(&furthest_point, p_b, point) {
+            left_pb.push(*point);
+        }
+    }
+    // recur
+    hull_set(p_a, &furthest_point, &mut left_ap, hull);
+    hull_set(&furthest_point, p_b, &mut left_pb, hull);
+}
+
+pub trait ConvexHull<T> {
+    /// Returns the convex hull of a Polygon
+    ///
+    /// This implementation uses the QuickHull algorithm,
+    /// based on [Barber, C. Bradford; Dobkin, David P.; Huhdanpaa, Hannu (1 December 1996)](https://dx.doi.org/10.1145%2F235815.235821)
+    /// Original paper here: http://www.cs.princeton.edu/~dpd/Papers/BarberDobkinHuhdanpaa.pdf
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Polygon};
+    /// use geo::convexhull::ConvexHull;
+    /// // an L shape
+    /// let coords = vec![(0.0, 0.0), (4.0, 0.0), (4.0, 1.0), (1.0, 1.0), (1.0, 4.0), (0.0, 4.0), (0.0, 0.0)];
+    /// let ls = LineString(coords.iter().map(|e| Point::new(e.0, e.1)).collect());
+    /// let poly = Polygon::new(ls, vec![]);
+    ///
+    /// // The correct convex hull coordinates
+    /// let hull_coords = vec![(0.0, 0.0), (0.0, 4.0), (1.0, 4.0), (4.0, 1.0), (4.0, 0.0), (0.0, 0.0)];
+    /// let correct_hull = LineString(hull_coords.iter().map(|e| Point::new(e.0, e.1)).collect());
+    ///
+    /// let res = poly.convex_hull();
+    /// assert_eq!(res.exterior, correct_hull);
+    /// ```
+    fn convex_hull(&self) -> Self where T: Float;
+}
+
+impl<T> ConvexHull<T> for Polygon<T>
+    where T: Float
+{
+    fn convex_hull(&self) -> Polygon<T> {
+        Polygon::new(LineString(quick_hull(&self.exterior.0)), vec![])
+    }
+}
```

Comment:
Seems like those can happen in follow up PRs, but would be good to have support for those other types